### PR TITLE
Tweak dependency handling

### LINKED ISSUE #178: No function correspond to methods exports for dbAppendTableArrow failing Docker install
When shiny app locally using pool, the following warning appears: 
Warning: no function found corresponding to methods exports from ‘pool’ for: ‘dbAppendTableArrow’, ‘dbCreateTableArrow’, ‘dbGetQueryArrow’, ‘dbReadTableArrow’, ‘dbSendQueryArrow’, ‘dbWriteTableArrow’.

For docker deployment using rocker/shiny-verse:4.2 as the starting image, this causes the package to not be installed and subsequent deployment issues.

### LINKED ISSUE #179: Error when using compute()  to create a new table in a schema.  Same code using connections created with dbConnect() works.
Hi everyone,
I am unable to use compute() to run a dbplyr query and save the output into a new table if that table is located in a schema if that connection was created using pool::dbPool().

EDIT: here is the error, to make it more obvious: 
```
# Error in UseMethod("sql_escape_ident") :
#   no applicable method for 'sql_escape_ident' applied to an object of class "c('Pool', 'R6')"


```

The code works if I create the connection using DBI::dbConnect()

Here's the closest I could create to a regexp.  It uses Azure (you won't be able to connect) because I am not sure how to create schemas in databases.

```R
#found in_schema(sql("schema"), "new_table ) in this post 
 #https://stackoverflow.com/questions/71410349/store-new-permanent-table-in-schema-using-compute


# libraries -----
library(dbplyr)
library(DBI)
library(pool)
library(dplyr)
library(odbc)
# connect using pool and con ----
pool <- pool::dbPool(drv = odbc::odbc(),
                     driver = "{ODBC17}",
                     uid = Sys.getenv("AZURE_UID"),          
                     pwd = Sys.getenv("AZURE_PWD"),        
                     server = Sys.getenv("AZURE_SERVER"),     
                     database = Sys.getenv("AZURE_DATABASE"),  
                     Authentication = "ActiveDirectoryPassword",
                     port = 1433,
                     minSize = 0, ## default is 1 which often leaves connections hanging (not cool)
                     idleTimeout = 600, ## 10 minutes until idle connection is closed, takes ~1 second to reopen
                     encoding = "UTF-8",
                     encrypt = "yes")

con <- DBI::dbConnect(
  odbc::odbc(),
  Driver   = "{ODBC Driver 17 for SQL Server}",
  Server   = Sys.getenv("AZURE_SERVER"),
  Database = Sys.getenv("AZURE_DATABASE"),
  UID      = Sys.getenv('AZURE_UID'),
  PWD      = Sys.getenv('AZURE_PWD'),
  Port     = 1433,
  Authentication = "ActiveDirectoryPassword",
  Encrypt  = "yes"
)



# upload iris to synapse  ----
#DBI::dbWriteTable(con, DBI::Id(schema = "my_schema",   table = "iris"),  value = iris, overwrite = TRUE) # slow but works

#  POOL --this doesnt work  -----

pool_iris <- tbl(pool,   dbplyr::in_schema(sql("my_schema"),"iris"))


pool_query <- pool_iris %>%
  head(10)

compute(pool_query, dbplyr::in_schema(sql("my_schema"),"iris_computed_from_pool"), temporary = FALSE)
# Error in UseMethod("sql_escape_ident") :
#   no applicable method for 'sql_escape_ident' applied to an object of class "c('Pool', 'R6')"


#  DBCONNECT - this works (same code) -----

con_iris <- tbl(con,   dbplyr::in_schema(sql("my_schema"),"iris"))


con_query <- con_iris %>%
  head(10)

compute(con_query, dbplyr::in_schema(sql("my_schema"),"iris_computed_from_con"), temporary = FALSE)


#  Source:   table<my_schema."iris_computed_from_con"> [10 x 5]
# # Database: Microsoft SQL Server 12.00.2531[usersqldatabaseXXXX@domainsqldatabaseXXXX.com@sqlserversqldatabaseXXXX/sqldatabaseXXXX]
#    Sepal.Length Sepal.Width Petal.Length Petal.Width Species
#           <dbl>       <dbl>        <dbl>       <dbl> <chr>
#  1          4.6         3.1          1.5         0.2 setosa
#  2          5.5         3.5          1.3         0.2 setosa
#  3          4.9         3.6          1.4         0.1 setosa
#  4          6.4         3.1          5.5         1.8 virginica
#  5          4.4         3.2          1.3         0.2 setosa



sessioninfo::session_info()
# ─ Session info ───────────────────────────────────────────────────────────────
#  setting  value
#  version  R version 4.0.2 (2020-06-22)
#  os       OpenShift Enterprise
#  system   x86_64, linux-gnu
#  ui       RStudio
#  language (EN)
#  collate  en_CA.UTF-8
#  ctype    en_CA.UTF-8
#  tz       America/Toronto
#  date     2024-02-06
#  rstudio  2023.06.0+421.pro1 Mountain Hydrangea (server)
#  pandoc   3.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/tools/ (via rmarkdown)
#
# ─ Packages ───────────────────────────────────────────────────────────────────
#  ! package     * version date (UTC) lib source
#  P bit           4.0.5   2022-11-15 [?] RSPM (R 4.0.5)
#  P bit64         4.0.5   2020-08-30 [?] RSPM (R 4.0.5)
#  P blob          1.2.4   2023-03-17 [?] RSPM (R 4.0.5)
#  P cli           3.6.0   2023-01-09 [?] RSPM (R 4.0.5)
#  P DBI         * 1.1.3   2022-06-18 [?] CRAN (R 4.0.5)
#  P dbplyr      * 2.3.2   2023-03-21 [?] RSPM (R 4.0.5)
#  P digest        0.6.31  2022-12-11 [?] RSPM (R 4.0.5)
#  P dplyr       * 1.1.0   2023-01-29 [?] RSPM (R 4.0.5)
#  P evaluate      0.20    2023-01-17 [?] RSPM (R 4.0.5)
#  P fansi         1.0.4   2023-01-22 [?] CRAN (R 4.0.2)
#  P fastmap       1.1.1   2023-02-24 [?] RSPM (R 4.0.5)
#  P generics      0.1.3   2022-07-05 [?] RSPM (R 4.0.5)
#  P glue          1.6.2   2022-02-24 [?] RSPM (R 4.0.5)
#  P hms           1.1.3   2023-03-21 [?] RSPM (R 4.0.5)
#  P htmltools     0.5.7   2023-11-03 [?] RSPM (R 4.0.5)
#  P knitr         1.42    2023-01-25 [?] RSPM (R 4.0.5)
#  P later         1.3.0   2021-08-18 [?] RSPM (R 4.0.5)
#  P lifecycle     1.0.3   2022-10-07 [?] RSPM (R 4.0.5)
#    magrittr      2.0.3   2022-03-30 [1] RSPM (R 4.0.5)
#  P odbc        * 1.4.2   2024-01-22 [?] RSPM (R 4.0.5)
#  P pillar        1.8.1   2022-08-19 [?] RSPM (R 4.0.5)
#  P pkgconfig     2.0.3   2019-09-22 [?] RSPM (R 4.0.3)
#  P pool        * 1.0.1   2023-02-21 [?] RSPM (R 4.0.5)
#  P purrr         1.0.1   2023-01-10 [?] RSPM (R 4.0.5)
#    R6            2.5.1   2021-08-19 [1] RSPM (R 4.0.5)
#  P Rcpp          1.0.10  2023-01-22 [?] CRAN (R 4.0.2)
#    renv          1.0.3   2023-09-19 [1] RSPM (R 4.0.2)
#  P rlang         1.1.0   2023-03-14 [?] RSPM (R 4.0.5)
#  P rmarkdown     2.20    2023-01-19 [?] RSPM (R 4.0.5)
#  P rstudioapi    0.15.0  2023-07-07 [?] RSPM (R 4.0.5)
#  P sessioninfo   1.2.2   2021-12-06 [?] RSPM (R 4.0.5)
#  P tibble        3.2.1   2023-03-20 [?] RSPM (R 4.0.5)
#  P tidyselect    1.2.0   2022-10-10 [?] RSPM (R 4.0.5)
#  P utf8          1.2.3   2023-01-31 [?] RSPM (R 4.0.5)
#  P vctrs         0.6.0   2023-03-16 [?] RSPM (R 4.0.5)
#  P withr         2.5.0   2022-03-03 [?] RSPM (R 4.0.5)
#  P xfun          0.37    2023-01-31 [?] RSPM (R 4.0.5)
#  P yaml          2.3.7   2023-01-23 [?] RSPM (R 4.0.5)
#
#  [1] /XXXX/renv/library/R-4.0/x86_64-pc-linux-gnu
#  [2] /opt/R/4.0.2/lib/R/library
#
#  P ── Loaded and on-disk path mismatch.

```

### PR DESCRIPTION
**Pull request by hadley**
* Bump required version of DBI. Fixes #178.
* Message if dbplyr is old. Fixes #179 (as much as possible)

### CODE REVIEW

File: tests/testthat/test-dbplyr.R [hunk 1]
```diff
@@ -91,3 +91,15 @@ test_that("wrapper looks good", {
...
+test_that("warns if dbplyr is old", {
+  local_mocked_bindings(packageVersion = function(...) "1.0.0")
+  expect_snapshot({
+    check_dbplyr()
+  })
+
+  cnd_dplyr <- class(catch_cnd(check_dbplyr()))
+  cnd_startup <- class(catch_cnd(packageStartupMessage("hi")))
+  cnd_startup <- c(cnd_startup[1:2], "rlang_message", cnd_startup[-(1:2)])
+  expect_equal(cnd_dplyr, cnd_startup)
```
Comment (lionel-) [inline tests/testthat/test-dbplyr.R:104]: This seems a bit brittle. Maybe do `skip_on_cran()` before?

**Review by lionel- — APPROVED** [review]
LG!

File: R/pool-package.R [hunk 2]
```diff
@@ -8,3 +8,8 @@
 #' @importFrom later later
 #' @import rlang
 NULL
+
+## usethis namespace: start
+#' @importFrom utils packageVersion
+## usethis namespace: end
```
Comment (DavisVaughan) [inline R/pool-package.R:14]: Might be worth cleaning up the imports in a follow up commit, just for future you

Comment (hadley) [reply R/pool-package.R:14]: You mean the other imports?

Comment (DavisVaughan) [reply R/pool-package.R:14]: yea just moving them to the usethis managed section is all i meant

**Pull request merged**